docs/extending-svd-viewer.md

```diff
@@ -0,0 +1,77 @@
+# Extending SVD Viewer
+
+It is possible to extend the SVD Viewer and register new file extension providers with your VSCode extension. This method will provide reading non-svd formats and load the peripherals information in SVD Viewer. 
+
+## Building your VSCode Extension to extend SVD Viewer
+
+This is a guide about how you can register new peripheral providers to SVD Viewer in your VSCode extension. Please refer to [VSCode Extension API](https://code.visualstudio.com/api) for more information about developing VSCode extensions.
+
+### Adding peripheral-viewer to your VSCode extension
+
+You need to install eclipse-cdt-cloud/vscode-svd-viewer to access the types information. You can use `npm` or `yarn` with the following arguments described below:
```

Comment:
A better approach would be to publish a small package containing the extension API export which can be consumed, would be a bit more work though.

This is what Arm do for the cmsis extension:

https://www.npmjs.com/package/@arm-software/vscode-cmsis-csolution